packages/amplify-cli/src/init-steps/s0-analyzeProject.ts

```diff
@@ -297,3 +299,29 @@ function getDefaultEditor() {
 
   return localEnvInfo.defaultEditor;
 }
+
+/**
+ *Checks if `amplify env add` has been executed, and extract environment name if exists
+ * @param {$TSContext} context The Amplify context object
+ * @returns `{exec: boolean, envName?: string | null}`
+ *
+ * `exec` denotes if `amplify env add` has been executed
+ *
+ * `envName` extracts environment name if provided
+ */
+function addNewEnv(context): { exec: boolean; envName?: string | null } {
```

Comment:
It seems like this function can just return a boolean (the value of `exec`), and the caller can use `context.parameters.first` to check for the environment name.

Reply:
Resolved!